frontend/dogplus-frontend/src/pages/userPage.tsx

```diff
@@ -1,55 +1,144 @@
-          method: 'GET',
+  const [refreshKey, setRefreshKey] = useState<number>(0);
+
+  const handleSearchToggle = () => {
+    setShowSearch(!showSearch);
+  };
+
+  const handleSearch = async () => {
+    const response = await fetch(
+      `${process.env.REACT_APP_BACKEND_HOST}/api/social/search/?username=${usernameToFollow}`,
+      {
+        headers: {
+          Authorization: `Token ${localStorage.getItem("token")}`,
+        },
+      }
+    );
+    if (response.ok) {
+      const data = await response.json();
+      setSearchResults(data.users);
+    }
+  };
+
+  const handleFollow = async (userId: string) => {
+    const response = await fetch(
+      `${process.env.REACT_APP_BACKEND_HOST}/api/social/follow/${userId}/`,
+      {
+        method: "POST",
+        headers: {
+          Authorization: `Token ${localStorage.getItem("token")}`,
+        },
+      }
+    );
+    if (response.ok) {
+      setSearchResults([]);
+      setRefreshKey((prev) => prev + 1);
+    }
+  };
+
+  const handleLogout = async () => {
+    try {
+      const response = await fetch(
+        `${process.env.REACT_APP_BACKEND_HOST}/api/auth/logout/`,
+        {
+          method: "POST",
           headers: {
-            Authorization: `Token ${localStorage.getItem('token')}`,
+            Authorization: `Token ${localStorage.getItem("token")}`,
           },
         }
       );
+
       if (response.ok) {
-        const data = await response.json();
-        setUser(data);
+        // Assuming the token is stored in localStorage; adjust if using cookies or other methods
+        localStorage.removeItem("token");
+        navigate("/auth"); // Redirect to login page after logout
       }
-    };
-    fetchUser();
-  }, []);
+    } catch (error) {
+      console.error("Logout failed", error);
+    }
+  };
 
   if (!user) {
     return <Loading />;
   }
 
   return (
-<div className="m-4 h-full flex flex-col">
-  <div>
-    <div className="flex items-center mb-2">
-      <img className="w-12 h-12 object-cover rounded-full mr-3" src={user.profile_image} alt="Profile Image" />
-      <div>
-        <h2 className="text-2xl font-semibold">{user.username}</h2>
+    <div className="m-4">
+      <div className="flex items-center mb-2">
+        <img
+          className="w-12 h-12 rounded-full mr-3"
+          src={user?.profile_image}
+          alt="Profile Image"
+        />
+        <h2 className="text-2xl font-semibold">{user?.username}</h2>
+        <button
+          type="button"
+          className="ml-auto text-foreground hover:text-white border bg-accent-0 hover:bg-accent-800 focus:ring-4 focus:outline-none focus:ring-accent-300 font-medium rounded-lg text-sm px-5 py-2.5 text-center"
+          onClick={() =>
+            navigate(`/user/edit/${localStorage.getItem("user_id")}`)
+          }
+        >
+          <FontAwesomeIcon icon={faEdit} size="lg" />
```

Comment:
In other parts of the code we use font awesome by having an element like:

```html
<div className="fas fa-edit">
```
This is tested on phones and we know it works. I believe it would be nice to have the entire frontend follow the same conventions regarding this